src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Windows/CertificatePal.PrivateKey.cs

```diff
@@ -80,6 +82,104 @@ public ECDsa GetECDsaPrivateKey()
             );
         }
 
+        public ICertificatePal CopyWithPrivateKey(DSA dsa)
+        {
+            DSACng dsaCng = dsa as DSACng;
+            ICertificatePal clone = null;
+
+            if (dsaCng != null)
+            {
+                clone = CopyWithPersistedCngKey(dsaCng.Key);
+
+                if (clone != null)
+                {
+                    return clone;
+                }
+            }
+
+            DSACryptoServiceProvider dsaCsp = dsa as DSACryptoServiceProvider;
+
+            if (dsaCsp != null)
+            {
+                clone = CopyWithPersistedCapiKey(dsaCsp.CspKeyContainerInfo);
+
+                if (clone != null)
+                {
+                    return clone;
+                }
+            }
+
+            DSAParameters privateParameters = dsa.ExportParameters(true);
```

Comment:
Yep. I know it's not totally guaranteed, but it has a pretty high likelihood of keeping most key material out of memory most of the time.